types/bitflags/bitflags32.ts

```diff
@@ -31,10 +33,11 @@ export class BitFlags32<
       }
     }
```

Comment:
I wonder if our types should increment the `byteOffset` after being read and written? Looking at https://github.com/denosaurs/byte_type/pull/13/files#diff-d0c4379f35f98a28f7ea97c21d11212d90d176bed83409ee594dce857aa4ed86R34 we did just that although maybe only structs and arrays (structured types? whatever we call those kinda types).

The argument for incrementing the `byteOffset` is of course that we could have an unsized struct or something similar containing unsized yet ordered content. Like a tuple of two null-terminated strings for example. In that case decoding would go happen like:
1. Unsized array of length two calls it's `element` types `NullTerminatedString.read` method with a reference to the type options `{ byteOffset: 0 }`
2. The first element reads until it encounters a `\0` null element. Increments the `byteOffset` to that index, say 13.
3. The array continues and does the same for the next element. Let's say this string is 4 bytes long including its null terminator.
5. It is returned to the user. If the user want's to know the `byteLength` of the now read array they can simply look at the `byteOffset` they may have passed to the inital read method call. In this example it will be 13 + 4 bytes = 17 bytes.

This makes it quite messy and easy to do off-by-one errors, but the payoff is performant and powerful reading/writing of unsized types.

Reply:
As discussed on discord. My next pr will add some inheritance classes to byte_type which should resolve this issue and simplify the design of components